src/cli/task.rs

```diff
@@ -273,15 +282,60 @@ pub fn execute(args: Args) -> miette::Result<()> {
             );
         }
         Operation::List(args) => {
-            let environment = project.environment_from_name_or_env_var(args.environment)?;
-            let tasks = environment
-                .tasks(Some(Platform::current()))?
-                .into_keys()
-                .collect_vec();
-            if tasks.is_empty() {
+            let explicit_environment = args
+                .environment
+                .map(|n| EnvironmentName::from_str(n.as_str()))
+                .transpose()?
+                .map(|n| {
+                    project
+                        .environment(&n)
+                        .ok_or_else(|| miette::miette!("unknown environment '{n}'"))
+                })
+                .transpose()?;
+            let available_tasks: HashSet<TaskName> =
+                if let Some(explicit_environment) = explicit_environment {
+                    explicit_environment
+                        .tasks(Some(Platform::current()))
+                        .into_iter()
+                        .flat_map(|tasks| {
+                            tasks.into_iter().filter_map(|(key, _)| {
+                                if !key.as_str().starts_with('_') {
+                                    Some(key)
+                                } else {
+                                    None
+                                }
+                            })
+                        })
+                        .map(ToOwned::to_owned)
+                        .collect()
```

Comment:
You could possibly move this into a function which you can reuse. e.g. `get_filtered_tasks` and reused that in the next section as well. And if you implement it for environment you could possibly also reuse it in `info`